infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -341,6 +345,9 @@ public void cleanShutdown() {
         if (replicationDiscoveryService != null) {
             replicationDiscoveryService.shutdown();
         }
+        // Flush the async appender before exiting to prevent the loss of logs
```

Comment:
Should we add a check to see if the `activeServer` is null?
If `startDiscoveryService` throws an exception, activeServer will be null and the shutdown hook hit NPE. Logger flush will be skipped.

https://github.com/CorfuDB/CorfuDB/blob/fa011a6f95c1bf586d9f8f22509e312fa3bd52bf/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java#L344

Reply:
done. thanks